core/src/main/scala/kafka/utils/Logging.scala

```diff
@@ -17,106 +17,96 @@
 
 package kafka.utils
 
-import org.apache.log4j.Logger
+import com.typesafe.scalalogging.{LazyLogging, Logger}
+import org.slf4j.{Marker, MarkerFactory}
 
-trait Logging {
-  val loggerName = this.getClass.getName
-  lazy val logger = Logger.getLogger(loggerName)
+import scala.util.{Failure, Success, Try}
 
-  protected var logIdent: String = null
+object Log4jControllerRegistration {
+  private val logger = Logger(this.getClass.getName)
 
-  // Force initialization to register Log4jControllerMBean
-  private val log4jController = Log4jController
+  val FatalMarker: Marker = MarkerFactory.getMarker("FATAL")
 
-  protected def msgWithLogIdent(msg: String) =
-    if (logIdent == null) msg else logIdent + msg
-
-  def trace(msg: => String): Unit = {
-    if (logger.isTraceEnabled())
-      logger.trace(msgWithLogIdent(msg))
-  }
-  def trace(e: => Throwable): Any = {
-    if (logger.isTraceEnabled())
-      logger.trace(logIdent,e)
+  val instance = Try {
```

Comment:
Why are we storing this instance? Seems like a plain `try/catch` would do the job.

Reply:
Indeed, I'll reformat it. Just thought it's more idiomatic to use `Try` and expression matches.

Reply:
`Try` is way more expensive so we generally only use in cases where its additional functionality is needed.

Reply:
I see, I'll rewrite it then.